simtools/layout/geo_coordinates.py

```diff
@@ -0,0 +1,163 @@
+import logging
+
+import numpy as np
+import pyproj
+
+
+class GeoCoordinates:
+    """
+    Geospatial Coordinate systems.
+    Defines UTM, WGS84 and local (CORSIKA) coordinate systems.
+
+    """
+
+    def __init__(self):
+        """
+        Initialize GeoCoordinates
+
+        """
+
+        self._logger = logging.getLogger(__name__)
+
+    def crs_utm(self, epsg):
+        """
+        UTM coordinate system definition.
+
+        Parameters
+        ----------
+        epsg: int
+            EPSG code for UTM zone.
+
+        Returns
+        -------
+        pyproj.CRS
+            UTM coordinate system.
+
+        """
+        if epsg:
+            crs_utm = pyproj.CRS.from_user_input(epsg)
+            self._logger.debug(f"UTM coordinate system: {crs_utm}")
+            return crs_utm
+
+        return None
+
+    @staticmethod
+    def crs_wgs84():
+        """
+        WGS84 coordinate system definition.
+
+        Returns
+        -------
+        pyproj.CRS
+            WGS84 coordinate system.
+
+        """
+        return pyproj.CRS("EPSG:4326")
+
+    def crs_local(self, reference_point):
+        """
+        Local coordinate system definition.
+
+        Parameters
+        ----------
+        reference_point: simtools.layout.telescope_position
+            Reference coordinate.
+
+        Returns
+        -------
+        pyproj.CRS
+            local coordinate system.
+
+        """
+        try:
+            _center_lat, _center_lon, _ = reference_point.get_coordinates("mercator")
+            _scale_factor_k_0 = self._coordinate_scale_factor(reference_point)
+            if not np.isnan(_center_lat.value) and not np.isnan(_center_lon.value):
```

Comment:
Should you add an `else` here to protect against the case one of them is `nan`? I think that without it you would simply return `None`, which might be OK, or perhaps you would want to raise an error like the missing reference point one.

Reply:
Good point. I've replace the if-statement by a try..expect pyproj.exceptions.CRSError: statement.

Reply:
I've reverted the changes back to simply return 'None' in case of `nan` inputs. This is the expected behavior of the function and there is not need for an exception.